components/zcash_note_encryption/src/lib.rs

```diff
@@ -743,3 +751,13 @@ pub fn try_output_recovery_with_ock<D: Domain, Output: ShieldedOutput<D, ENC_CIP
         None
     }
 }
+
+// Splits the AEAD tag from the actual ciphertext .
+fn extract_tag(enc_ciphertext: &mut Vec<u8>) -> (&mut [u8], [u8; AEAD_TAG_SIZE]) {
+    let tag_loc = enc_ciphertext.len() - AEAD_TAG_SIZE;
+
+    let (plaintext, tail) = enc_ciphertext.split_at_mut(tag_loc);
```

Comment:
Do we want to name it something other than `plaintext` for clarity? Since it has not been decrypted.